frontend/src/__tests__/cypress/cypress/pages/workbench.ts

```diff
@@ -0,0 +1,352 @@
+import { Contextual } from './components/Contextual';
+import { Modal } from './components/Modal';
+import { TableRow } from './components/table';
+
+class StorageModal extends Modal {
+  constructor() {
+    super('Add storage to Test Notebook');
+  }
+
+  selectExistingPersistentStorage(name: string) {
+    this.find()
+      .findByTestId('persistent-storage-group')
+      .findByRole('button', { name: 'Options menu' })
+      .findSelectOption(name)
+      .click();
+  }
+
+  findSubmitButton() {
+    return this.find().findByTestId('attach-storage');
+  }
+
+  findWorkbenchRestartAlert() {
+    return this.find().findByTestId('notebook-restart-alert');
+  }
+
+  findMountField() {
+    return this.find().findByTestId('mount-path-folder-value');
+  }
+}
+class WorkbenchPage {
+  visit(projectName: string) {
+    cy.visitWithLogin(`/projects/${projectName}?section=workbenches`);
```

Comment:
Use `cy.visit` now because of https://github.com/opendatahub-io/odh-dashboard/pull/2682